core/src/worker/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -65,21 +126,39 @@ pub(super) struct Created {}
 /// higher-level machinery, it transitions into this state.
 #[derive(Debug, Default, Clone, derive_more::Display)]
 pub(super) struct CommandIssued {}
+pub(super) struct CmdRecDat {
+    sa_map: Option<SearchAttributes>,
+    replaying: bool,
+}
+
+impl CommandIssued {
+    pub(super) fn on_command_recorded(
+        self,
+        shared: SharedState,
+        dat: CmdRecDat,
+    ) -> UpsertSearchAttributesMachineTransition<Done> {
+        if shared.internal_flags.borrow_mut().try_use(
+            CoreInternalFlags::UpsertSearchAttributeOnPatch,
+            !dat.replaying,
```

Comment:
Wouldn't the only time to check this be on replay? Or am I misunderstanding this field? Can you confirm there is a test testing this nondeterminism error?

Reply:
There is a test that verifies this works. The replay flag is false the first time you get a new WFT with the recorded command event. Replay only becomes true once we've caught up to the first non-command event in the most recent WFT.